curvlinops/kfac.py

```diff
@@ -555,6 +563,18 @@ def _compute_loss_and_backward(self, output: Tensor, y: Tensor):
             for mc in range(self._mc_samples):
                 y_sampled = self.draw_label(output)
                 loss = self._loss_func(output, y_sampled)
+
+                if (
+                    isinstance(self._loss_func, (BCEWithLogitsLoss, MSELoss))
+                    and self._loss_func.reduction == "mean"
+                ):
+                    # ``BCEWithLogitsLoss`` and ``MSELoss`` also average over non-batch
+                    # dimensions. We have to scale the loss to incorporate this scaling
+                    # as we cannot generally achieve it by incorporating it into the
+                    # drawn sample.
+                    C = output.shape[1:].numel()
```

Comment:
Should `output.shape[1]` be sufficient here since we always reshape the outputs to be 2d?

Reply:
Made more explicit.